### PR TITLE
Adds GLOBE 3D + Dual Tree Traversal Accelerations 🌎🌲 

### PR DESCRIPTION
**Pull request by peterdsharpe**
# PhysicsNeMo Pull Request

## Description

This is a really fun PR that includes:

- A new GLOBE example, set up for surface prediction on the (3D) DrivAerML automotive aerodynamics case
- GLOBE model architecture modifications that fundamentally break the previous O(n^2) bottleneck, enabling scaling to large industrial problems. Scaling is now (both theoretically and empirically verified) to be O(n), which is very exciting to see for an all-to-all interaction problem. This is performed by doing a dual-tree variant of classical Barnes-Hut (or Fast-Multipole-Method-like) hierarchical acceleration by exploiting spatial locality. Normally Barnes-Hut would get you to O(n log n), but it turns out that by putting both the sources AND the targets in trees (a "dual tree traversal" approach), you can make it so that the interactions are no longer the bottleneck at all. This leaves O(n) operations as the bottleneck.
    - This dual-tree algorithm is implemented using a variant of a Linear Bounding Volume Hierarchy (LBVH), which is a nice way to make this more GPU-friendly (basically, this allows each layer of the tree (or in the case of dual tree traversal, each combination of layers from both trees) to coalesce into a single kernel launch, rather than doing a full tree traversal for every interaction.

Adds a theory doc about the dual tree traversal approach at `./physicsnemo/experimental/models/globe/hierarchical_acceleration.md`.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [x] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/experimental/models/globe/field_kernel.py [hunk 1]
```diff
@@ -23,12 +23,14 @@
...
-import tqdm
 from jaxtyping import Float
 from tensordict import TensorDict
+from torch.profiler import record_function
 from torch.utils.checkpoint import checkpoint
 
 from physicsnemo.core.module import Module
+
+logger = logging.getLogger("globe.field_kernel")
 from physicsnemo.experimental.models.globe.utilities.rank_spec import (
```
[the remark this answers is not in the document]
Comment (peterdsharpe) [reply physicsnemo/experimental/models/globe/field_kernel.py:32]: Fixed

File: examples/cfd/external_aerodynamics/globe/drivaer/train.py [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+        keys = ["loss", *all_batch_loss_components.keys()]
+        all_values = torch.stack(
+            [
+                torch.nanmean(torch.stack(all_batch_losses)),
+                *(
+                    torch.nanmean(torch.stack(all_batch_loss_components[k]))
+                    for k in keys[1:]
+                ),
+            ]
+        )
```
[the remark this answers is not in the document]
Comment (peterdsharpe) [reply examples/cfd/external_aerodynamics/globe/drivaer/train.py:458]: This is an edge case that can't happen on this example. But, I did add a len != 0 guard in the dataset, just in case.

**Comment by coreyjadams** [comment]
Hey @peterdsharpe ....

We are not in a time crunch here nor are we rushing this one out the door right.  What do you think about, you know, breaking this PR up into more manageable chunks for review?

Perhaps we can target the barnes hut work as the first layer standalone?  Then do some mesh updates?  Then bring in the 3D globe model?

It would make the review burden a lot easier?

**Comment by peterdsharpe** [comment]
Sounds good, will do!
